tests/views/test_collection_exercise.py

```diff
@@ -198,503 +142,260 @@ class File:
+    @requests_mock.mock()
+    def test_set_live_button(self, mock_request):
+        sign_in_with_permission(self, mock_request, user_permission_surveys_edit_json)
         response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
 
         self.assertEqual(response.status_code, 200)
         self.assertIn("Monthly Survey of Building Materials Bricks".encode(), response.data)
         self.assertIn("221_201712".encode(), response.data)
         self.assertIn("Set as ready for live".encode(), response.data)
-        self.assertNotIn("Pre-Populated data is available for this sample".encode(), response.data)
 
+    @mock_decorator(CE_READY, EQ_SURVEY, EQ_CI, SAMPLE_SUMMARY, {})
+    def test_set_live_button_invalid(self):
+        response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertNotIn("Set as ready for live".encode(), response.data)
+
+    @mock_decorator(CE_READY, EQ_SURVEY, EQ_CI, SAMPLE_SUMMARY, CE_EVENTS)
     @requests_mock.mock()
     @patch("response_operations_ui.controllers.collection_instrument_controllers.get_response_json_from_service")
-    def test_collection_exercise_view_cir_count_valid(self, mock_request, registry_instrument_count):
+    def test_cir_count_valid(self, mock_request, registry_instrument_count):
+        sign_in_with_permission(self, mock_request, user_permission_surveys_edit_json)
         self.app.config["CIR_ENABLED"] = True
         registry_instrument_count.return_value = {"registry_instrument_count": 1}
-        self.load_eq_survey(
-            mock_request,
-            self.eq_survey_dates,
-            self.collection_exercises,
-            collection_exercise_eq_both_ref_date["collection_exercise"],
-            self.collection_exercise_ref_both_date,
-            sample_summary_id,
-            self.sample_summary,
-            self.eq_collection_instrument,
-            self.eq_ci_selectors,
-        )
-        mock_request.get(url_get_by_survey_with_ref_start_date, json=collection_exercise_eq_ref_start_date)
-        mock_request.get(url_get_by_survey_with_ref_end_date, json=collection_exercise_eq_ref_end_date)
 
         response = self.client.get(f"/surveys/{short_name}/{period}", follow_redirects=True)
+
         self.assertIn("Set as ready for live".encode(), response.data)
 
-    @requests_mock.mock()
+    @mock_decorator(CE_READY, EQ_SURVEY, EQ_CI, SAMPLE_SUMMARY, CE_EVENTS)
     @patch("response_operations_ui.controllers.collection_instrument_controllers.get_response_json_from_service")
-    def test_collection_exercise_view_cir_count_invalid(self, mock_request, registry_instrument_count):
+    def test_cir_count_invalid(self, registry_instrument_count):
         self.app.config["CIR_ENABLED"] = True
         registry_instrument_count.return_value = {"registry_instrument_count": 0}
-        self.load_eq_survey(
```

Comment:
A great example of my point above, so much going on, making the test look complicated and difficult to manage, the decorator quietens the noise. Fixtures will make this even better